src/test/java/com/google/sps/data/ScheduledInterviewServletTest.java

```diff
@@ -0,0 +1,184 @@
+    MockHttpServletRequest postRequest = new MockHttpServletRequest();
+
+    postRequest.addParameter("startTime", "2020-07-05T18:00:00Z");
+    postRequest.addParameter("endTime", "2020-07-05T19:00:00Z");
+    postRequest.addParameter("interviewer", "user@company.org");
+    postRequest.addParameter("interviewee", "user@gmail.com");
+    scheduledInterviewServlet.doPost(postRequest, new MockHttpServletResponse());
+
+    postRequest.setParameter("startTime", "2020-07-05T20:00:00Z");
+    postRequest.setParameter("endTime", "2020-07-05T21:00:00Z");
+    postRequest.setParameter("interviewer", "user2@company.org");
+    postRequest.setParameter("interviewee", "user@gmail.com");
+    scheduledInterviewServlet.doPost(postRequest, new MockHttpServletResponse());
+
+    postRequest.setParameter("startTime", "2020-07-05T18:00:00Z");
+    postRequest.setParameter("endTime", "2020-07-05T19:00:00Z");
+    postRequest.setParameter("interviewer", "user2@company.org");
+    postRequest.setParameter("interviewee", "user1@gmail.com");
+    scheduledInterviewServlet.doPost(postRequest, new MockHttpServletResponse());
+
+    getRequest.addParameter("userEmail", "user@gmail.com");
+    scheduledInterviewServlet.doGet(getRequest, getResponse);
+
+    Type scheduledInterviewListType =
+        new TypeToken<ArrayList<ScheduledInterviewRequest>>() {}.getType();
+    JsonElement json = new JsonParser().parse(getResponse.getContentAsString());
+    List<ScheduledInterviewRequest> actual = new Gson().fromJson(json, scheduledInterviewListType);
+
+    ScheduledInterviewRequest scheduledInterview1 =
+        new ScheduledInterviewRequest(
+            actual.get(0).getId(),
+            new TimeRange(
+                Instant.parse("2020-07-05T18:00:00Z"), Instant.parse("2020-07-05T19:00:00Z")),
+            "user@company.org",
+            "user@gmail.com");
+    ScheduledInterviewRequest scheduledInterview2 =
+        new ScheduledInterviewRequest(
+            actual.get(1).getId(),
+            new TimeRange(
+                Instant.parse("2020-07-05T20:00:00Z"), Instant.parse("2020-07-05T21:00:00Z")),
+            "user2@company.org",
+            "user@gmail.com");
+
+    List<ScheduledInterviewRequest> expected = new ArrayList<ScheduledInterviewRequest>();
+    expected.add(scheduledInterview1);
+    expected.add(scheduledInterview2);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Tests errors with Instant parsing.
+  @Test
+  public void bad() throws IOException {
+    ScheduledInterviewServlet scheduledInterviewServlet = new ScheduledInterviewServlet();
+    scheduledInterviewServlet.init(new FakeScheduledInterviewDao());
+    helper.setEnvIsLoggedIn(true).setEnvEmail("user@company.org").setEnvAuthDomain("auth");
+    MockHttpServletRequest postRequest = new MockHttpServletRequest();
+    MockHttpServletResponse postResponse = new MockHttpServletResponse();
+    postRequest.addParameter("startTime", "22020-07-0518:00:00Z");
```

Comment:
Probably change the "22020" year back to "2020" so there is only one invalid part of the date (also 22020 may very well be considered a valid year).

Reply:
Done